src/board/board.ts

```diff
@@ -24,3 +35,28 @@ export const renderBoard = (data: Board) => {
   });
   console.log(chalk.white.bold("-------------"));
 };
+
+export const updateBoardState = (
+  board: Board,
+  action: UserInput,
+  gameState: GameState
+): Board => {
+  const { x: lastPlayedLineIndex, y: lastPlayedTileIndex } =
+    findLastPLayed(board);
+
+  if (lastPlayedLineIndex != undefined && lastPlayedTileIndex != undefined) {
+    board[lastPlayedLineIndex][lastPlayedTileIndex] = removeLastPlayed(
+      board[lastPlayedLineIndex][lastPlayedTileIndex] as PlayableTile
+    );
+  }
+
+  const { x: lineIndex, y: tileIndex } = findTile(board, action.value);
+  const tile = playTile(
+    findTileByCoordinate(board, { x: lineIndex, y: tileIndex }),
+    gameState.currentPlayer
+  );
+
+  board[lineIndex][tileIndex] = tile;
+
+  return [...board];
```

Comment:
Not sure why you're returning `[...board]` and not just `board`? :thinking:

Reply:
A misunderstanding. Done